src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/step/rollup/WaitForRollupCompletionStep.kt

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.indexmanagement.indexstatemanagement.step.rollup
+
+import com.amazon.opendistroforelasticsearch.indexmanagement.elasticapi.suspendUntil
+import com.amazon.opendistroforelasticsearch.indexmanagement.indexstatemanagement.model.ManagedIndexMetaData
+import com.amazon.opendistroforelasticsearch.indexmanagement.indexstatemanagement.model.managedindexmetadata.StepMetaData
+import com.amazon.opendistroforelasticsearch.indexmanagement.indexstatemanagement.step.Step
+import com.amazon.opendistroforelasticsearch.indexmanagement.rollup.action.explain.ExplainRollupAction
+import com.amazon.opendistroforelasticsearch.indexmanagement.rollup.action.explain.ExplainRollupRequest
+import com.amazon.opendistroforelasticsearch.indexmanagement.rollup.action.explain.ExplainRollupResponse
+import com.amazon.opendistroforelasticsearch.indexmanagement.rollup.model.RollupMetadata
+import org.apache.logging.log4j.LogManager
+import org.elasticsearch.client.Client
+import org.elasticsearch.cluster.service.ClusterService
+import java.lang.Exception
+
+class WaitForRollupCompletionStep(
+    val clusterService: ClusterService,
+    val client: Client,
+    managedIndexMetaData: ManagedIndexMetaData
+) : Step(name, managedIndexMetaData) {
+
+    private var stepStatus = StepStatus.STARTING
+    private var info: Map<String, Any>? = null
+    private val logger = LogManager.getLogger(javaClass)
+    private var hasRollupFailed: Boolean? = null
+
+    override fun isIdempotent() = true
+
+    override suspend fun execute(): WaitForRollupCompletionStep {
+        val rollupJobId = managedIndexMetaData.actionMetaData?.actionProperties?.rollupId
+
+        if (rollupJobId == null) {
+            logger.error("No rollup job id passed down")
+            stepStatus = StepStatus.FAILED
+        } else {
```

Comment:
Is this a real use case? If not could we not just pass down the config into the step that already contains the rollup ID?

Reply:
I can't think of a case this will happen but programmatically it can since its an optional field. Can pass it down through the input the instead but would need to pass the whole `ISMRollup` object

Reply:
Ok, can either update to use the ISMRollup or just update this to at least include a message in info object.

Reply:
added the info message